packages/common/src/errors.ts

```diff
@@ -52,3 +54,18 @@ export class NamespaceNotFoundError extends Error {
     super(`Namespace not found: '${namespace}'`);
   }
 }
+
+/**
+ * Thrown when it's necessary to rethrow a grpc-js ServiceError with modifications.
+ */
+@SymbolBasedInstanceOfError('CustomGrpcServiceError')
+export class CustomGrpcServiceError extends Error {
```

Comment:
Put that error in `client/src/errors.ts` instead.

Reply:
Also, not a fan of the name. "custom" in this context means nothing to the end user. Could we have more targetted errors? Or merge this one with `client.ServiceError`?

Reply:
Reverted all changes to `common/src`. Still not sure how to rethrow `GrpcServiceError`. One option seems to be just to mutate the received error and re-throw it.

Reply:
I am:

- Identifying `WorkflowExecutionAlreadyStartedError` and using that as the error if appropriate
- Otherwise mutating the received error so that it acquires the more specific GrpcServiceError fields (code, message) as unpacked from the MultiOp error, and rethrowing the altered GrpcServiceError error